app/models/card.rb

```diff
@@ -3,29 +3,59 @@ class Card < ActiveRecord::Base
   belongs_to :user
   accepts_nested_attributes_for :deck
 
-  before_validation :set_default_review_date
+  before_create :set_defaults
+
+  LEITNER_TIME = [
+    0,
+    12.hours,
+    3.days,
+    1.week,
+    2.weeks,
+    1.month
+  ].freeze
+
+  REVIEW_LEVELS = LEITNER_TIME.length - 1
+  ERRORS_TO_DECREASE = 3
 
   has_attached_file :picture, styles: { medium: "360x360>", thumb: "50x50>" }, default_url: "/images/placeholder.png"
   validates_attachment_content_type :picture, content_type: /\Aimage\/.*\Z/
-  validates :original_text, :translated_text, :review_date, presence: true
+  validates :original_text, :translated_text, presence: true
   validates :deck_id, presence: true
   validate :not_equal_fields
-  scope :for_review, -> { where("review_date < ?", Date.today) }
-
-  def move_review_date
-    self.review_date = Date.today + 3.days
-    save
-  end
+  scope :for_review, -> { where("review_date < ?", Time.now) }
 
   def check_translation(translation)
     if translated_text == clear(translation)
-      move_review_date
+      increase_review_level
       return true
     else
+      decrease_review_level
       return false
     end
   end
 
+  def increase_review_level
+    self.error_counter = 0
+    if review_level < REVIEW_LEVELS
+      self.review_level += 1
+      new_date = Time.now + LEITNER_TIME[review_level]
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.